### PR TITLE
pkg: quiet pkg builds

### PR DESCRIPTION
**Pull request by kaspar030**
This PR

- prepends $(Q) to most pkg/pkg.mk lines
- adds a shell helper library containing a "quiet" helper in dist/tools/functions.sh
- quiets the relic build using that helper

### CODE REVIEW

**Comment by kaspar030** [comment]
> quiets the relic build using that helper

While this is conditional (QUIET=0 prints the output), a side effect is that the output is printed all at once, once the command (in this case the whole relic build) has completed.

File: dist/tools/functions.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+# run arguments as command, only print output when command errors
+quiet() {
+    OUT="$("$@" 2>&1)"
+    RES=$?
+    [ -n "$OUT" ] && [ $RES -ne 0 -o "$QUIET" = "0" ] && echo "$OUT"
+    return $RES
+}
+
+echo $QUIET
```
Comment (cgundogan) [inline dist/tools/functions.sh]: do we need to print that? In the logs it looks like a random "1" popping out of nowhere.

**Comment by cgundogan** [comment]
works! please see my comment though

File: dist/tools/functions.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+# run arguments as command, only print output when command errors
+quiet() {
+    OUT="$("$@" 2>&1)"
+    RES=$?
+    [ -n "$OUT" ] && [ $RES -ne 0 -o "$QUIET" = "0" ] && echo "$OUT"
```
Comment (cgundogan) [inline dist/tools/functions.sh]: wouldn't `"$QUET" -eq 0` also work here?

Comment (kaspar030) [inline dist/tools/functions.sh]: Using ```-eq``` fails if not both arguments are an integer. and as QUIET might be use supplied (e.g., QUIET=true), it is safer to to string comparison here.

See ```# test a -eq 0``` -> ```-bash: test: a: integer expression expected```

RES on the other hand is definitely filed with an integer value, so using ```-ne``` is fine.

Comment (cgundogan) [inline dist/tools/functions.sh]: but wouldn't this mean that if the user supplies `QUIET=false` it does not have the correct effect?

File: dist/tools/functions.sh (hunk 1 above)
Comment (kaspar030) [inline dist/tools/functions.sh]: ups, that's a debugging leftover. will fix.

Comment (kaspar030) [inline dist/tools/functions.sh]: did

File: dist/tools/functions.sh (hunk 2 above)
Comment (kaspar030) [inline dist/tools/functions.sh]: yes...

```examples/hello-world/README.md:* The variable `QUIET`, which is either `1` or `0`, defines whether to print verbose compile information, or hide them,
respectively.```

**Comment by kaspar030** [comment]
- rebased, squashed

**Comment by kaspar030** [comment]
Hm, somehow the build fails on Murdock. Locally everything works fine. ;(

**Comment by cgundogan** [comment]
https://riot-ci.inet.haw-hamburg.de/job/RIOT_PR/job/PR-6488/3/artifact/error_native_linux_unittests.log

@smlng AFAIK you encountered this error before, don't you?

**Comment by miri64** [comment]
Fails for some reason I can't find on Murdock with x86.

**Comment by miri64** [comment]
Maybe it works on current murdock?

**Comment by kaspar030** [comment]
> Maybe it works on current murdock?

Nothing changed since April.

**Comment by miri64** [comment]
Any idea why it's failing?

**Comment by smlng** [comment]
all failures seem to be `mips` related, I trigger Murdock again - just to be sure the issue persists.

**Comment by miri64** [comment]
@kaspar030 can you have a look?

**Comment by miri64** [comment]
Maybe, since it is MIPS-related, @neiljay as well?

**Comment by miri64** [comment]
@kaspar030 can you rebase? Maybe that already helps, since this PR seems to still be of Murdock2 vs Jenkins times it seems to me there is some pretty old code integrated

**Comment by kaspar030** [comment]
I found the problem. The relic build passes parameters like ```-DSOMETHING=AB;BB;CC;DD```, which wasn't properly escaped. I've added a helper function that takes care of that.

**Comment by kaspar030** [comment]
> I found the problem.

Unfrtunately not ;(

**Comment by miri64** [comment]
@cladmi you gathered some experience with both MIPS and our build system. Can you maybe figure out, what is happening?

**Comment by miri64** [comment]
Ping @cladmi?

**Comment by miri64** [comment]
Ping @kaspar030 :grinning:

**Comment by kaspar030** [comment]
Honestly I don't care anymore...

**Pull request closed without merging**